### ISSUE TITLE
The comma rules pass over every comma behind an address, taking its double slash for the start of a comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#153]
Found by an adversarial review of the branch that repairs #115, and unrelated to it: this needs no comment anywhere and no syntax but plain CSS.

`style-search`, which finds the commas of a value and of a media query list, reads a double slash as the opening of an inline comment wherever it stands — the one in `http://` among them. Everything from there to the end of the line is comment text as far as the search is concerned, the closing parenthesis of the `url()` with it, so the count of open parentheses never comes back down and `functionArguments: "skip"` drops every comma that follows.

### What minimal example or steps are needed to reproduce the bug?

```css
a { b: url(http://x/y.png),'z'; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/value-list-comma-space-before": "always"
  }
}
```

### What did you expect to happen?

A warning about the comma, which has no space in front of it. The same value with the two slashes taken out of the address draws one:

```css
a { b: url(x/y.png),'z'; }
```

### What actually happened?

Nothing at all. `value-list-comma-space-after` says nothing either, and so does every rule that goes through `valueListCommaWhitespaceChecker` or `mediaQueryListCommaWhitespaceChecker` for a value carrying such an address.

`functionCommaSpaceChecker` reads the same double slash the same way, through a pattern of its own rather than through `style-search`: it strips ` //` and the rest of the line out of the string it maps its indexes in, so `function-comma-space-before` set to `never` says nothing about `image-set(url(//cdn/a.png) 1x ,url(//cdn/b.png) 2x)`.

### Do you have a proposal to fix the bug?

The plugin already knows where a comment of that kind stands and where one only looks like it: `findInlineCommentSpans` passes over the double slash of an address, quoted or bare, and over one inside a block comment. Handing `style-search` a copy of the value in which those false comments are spelled out of harm's way — as the checkers repaired by #115 already hand it one with the real comments closed where the syntax closes them — would answer both halves.

The two readings inside `functionCommaSpaceChecker` should become one at the same time: it places its commas against `findInlineCommentSpans` and maps their positions with `COMMENT_PATTERN`, and the two do not agree about an address.
